ocs_ci/ocs/scale_lib.py

```diff
@@ -709,3 +709,94 @@ def increase_pods_per_worker_node_count(pods_per_node=500, pods_per_core=10):
+        access_mode (str): PVC access_mode
+        sc_name (str): SC name for pvc creation
+
+    Returns:
+         pvc_dict_list (list): List of all PVC.yaml dicts
+
+    """
+    if access_mode == 'ReadWriteMany' and 'rbd' in sc_name:
+        volume_mode_block_flag = 1
+    else:
+        volume_mode_block_flag = 0
+
+    # Construct PVC.yaml for the no_of_required_pvc count
+    # append all the pvc.yaml dict to pvc_dict_list and return the list
+    pvc_dict_list = list()
+    for i in range(no_of_pvc):
+        pvc_name = helpers.create_unique_resource_name('test', 'pvc')
+        size = f"{random.randrange(5, 105, 5)}Gi"
+        pvc_data = templating.load_yaml(constants.CSI_PVC_YAML)
+        pvc_data['metadata']['name'] = pvc_name
+        del pvc_data['metadata']['namespace']
+        pvc_data['spec']['accessModes'] = [access_mode]
+        pvc_data['spec']['storageClassName'] = sc_name
+        pvc_data['spec']['resources']['requests']['storage'] = size
+        # Check to identify RBD_RWX PVC and add VolumeMode
+        if volume_mode_block_flag:
+            pvc_data['spec']['volumeMode'] = 'Block'
+        else:
+            pvc_data['spec']['volumeMode'] = None
+        pvc_dict_list.append(pvc_data)
+
+    return pvc_dict_list
+
+
+def check_all_pvc_reached_bound_state_in_kube_job(kube_job_obj, namespace, no_of_pvc):
+    """
+    Function to check either bulk created PVCs reached Bound state using kube_job
+
+    Args:
+        kube_job_obj (obj): Kube Job Object
+        namespace (str): Namespace of PVC's created
+        no_of_pvc (int): Bulk PVC count
+
+    Returns:
+        pvc_bound_list (list): List of all PVCs which is in Bound state.
+
+    Asserts:
+        If not all PVC reached to Bound state.
+
+    """
+    # Check all the PVC reached Bound state
+    pvc_bound_list, pvc_not_bound_list = ([] for i in range(2))
+    while_iteration_count = 0
+    while True:
+        # Get kube_job obj and fetch either all PVC's are in Bound state
+        # If not bound adding those PVCs to pvc_not_bound_list
+        job_get_output = kube_job_obj.get(namespace=namespace)
+        for i in range(no_of_pvc):
+            status = job_get_output['items'][i]['status']['phase']
+            logging.info(f"pvc {job_get_output['items'][i]['metadata']['name']} status {status}")
+            if status != 'Bound':
+                pvc_not_bound_list.append(job_get_output['items'][i]['metadata']['name'])
+
+        # Check the length of pvc_not_bound_list to decide either all PVCs reached Bound state
+        # If not then wait for 30secs and re-iterate while loop
+        if len(pvc_not_bound_list):
+            time.sleep(30)
+            while_iteration_count += 1
+            # Breaking while loop after 10 Iteration i.e. after 30*10 secs of wait_time
+            # And if PVCs still not in bound state then there will be assert.
+            if while_iteration_count >= 10:
+                assert logging.error(
```

Comment:
it would be helpful to know PVCs that are not bound for better debugging.

Reply:
Its there in code 
```
            if while_iteration_count >= 10:
                assert logging.error(
                    f" Listed PVCs took more than 600secs to bound {pvc_not_bound_list}"
                )
                break
```

Reply:
When PVCs are not in Bound then we can find something like below in log output.
```
13:19:33 - MainThread - /home/rperiyas/Documents/tools/OCS4x/my_changes/testcase/kube-job/pvc-create/ocs-ci/ocs_ci/ocs/resources/objectconfigfile.py - INFO - going to run oc get on rbd_pvc_job object config yaml file in namespace namespace-test-f4b845ecac5a49979b650a1513673bbd
13:19:33 - MainThread - ocs_ci.utility.utils - INFO - Executing command: ['oc', '--kubeconfig', '/home/rperiyas/vmware/auth/kubeconfig', 'get', '-f', '/tmp/pytest-of-rperiyas/pytest-6/test_all_4_type_pvc_creation_d0/objectconfig.rbd_pvc_job.yaml', '-n', 'namespace-test-f4b845ecac5a49979b650a1513673bbd', '-o', 'yaml']
13:19:39 - MainThread - root - INFO - pvc pvc-test-3a00b46b16f440c3804a04bdfa407ba4 status Bound
13:19:39 - MainThread - root - INFO - pvc pvc-test-3668da59f5c64d85aab8b224d06e9c2c status Bound
13:19:39 - MainThread - root - INFO - pvc pvc-test-1c248b7e9e214e85892e02d6fe475aaa status Bound
13:19:39 - MainThread - root - INFO - pvc pvc-test-69b3f2d9ba7d4a5fa6df6634425ab40c status Bound
13:19:39 - MainThread - root - INFO - pvc pvc-test-1c7191ec7c674dd2878384a1f9dbd1df status Bound
13:19:39 - MainThread - root - INFO - pvc pvc-test-0650b350d82b4a2db90a3d59782c8f5d status Pending
13:19:39 - MainThread - root - INFO - pvc pvc-test-629a9fb6626f4aabb99a383a2778a9a9 status Pending
13:19:39 - MainThread - root - INFO - pvc pvc-test-fd6e827ba5ed4f2eaa9ec85f1f43564d status Pending
13:19:39 - MainThread - root - INFO - pvc pvc-test-6bc8279081ff49c7a447cbbf656b3a2f status Pending
13:19:39 - MainThread - root - INFO - pvc pvc-test-9245926f545849b2bf1a803d4b69adad status Pending
13:20:09 - MainThread - root - ERROR -  Listed PVCs took more than 300secs to bound ['pvc-test-0650b350d82b4a2db90a3d59782c8f5d', 'pvc-test-629a9fb6626f4aabb99a383a2778a9a9', 'pvc-test-fd6e827ba5ed4f2eaa9ec85f1f43564d', 'pvc-test-6bc8279081ff49c7a447cbbf656b3a2f', 'pvc-test-9245926f545849b2bf1a803d4b69adad']
```